clash: update to 1.3.0

eatradish opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

V2ray desktop and clash fix build

Package(s) Affected
-------------------

clash 1.3.0
(-) v2ray-desktop 2.1.7-1

Security Update?
----------------

No

<!-- Please uncomment the "Build Order" section if applicable, this is commonly needed in package updates/introduction that affects more than one package. -->


Architectural Progress
----------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Secondary Architectural Progress
--------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Post-Merge Secondary Architectural Progress
-------------------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

Nexus01 committed:
clash: fix version information

Nexus01 committed:
v2ray-desktop: build-fix

    - Enable clash detection.

eatradish committed:
Merge remote-tracking branch 'Nexus01/clash-v2d' into v2ray-desktop-and-clash-fix-build

eatradish committed:
clash: lint

eatradish committed:
v2ray-desktop: lint

eatradish committed:
clash: update to 1.3.0

MingcongBai commented on extra-network/clash/autobuild/build:
```diff
@@ -0,0 +1,5 @@
+abinfo "Building clash..."
+go build -trimpath -ldflags "-X github.com/Dreamacro/clash/constant.Version=$PKGVER" -mod=readonly
```
How did `ABTYPE=gomod` fail?

eatradish replied:
The `-trimpath` argument is required here, otherwise the `--version` argument of the compiled clash binary cannot display the version.

MingcongBai replied:
Got it, Autobuild3 does not yet handle these scenarios well.

MingcongBai commented on extra-network/v2ray-desktop/autobuild/prepare:
```diff
@@ -1 +1,4 @@
 acbs_copy_git
+abinfo "Set the v2ray_usr_local_install..."
```
This `abinfo` is not descriptive. It also does not address the second `sed` operation.

MingcongBai commented on extra-network/v2ray-desktop/autobuild/beyond:
```diff
@@ -1,9 +1,14 @@
+abinfo "Replacing Exec path..."
+sed -i 's|%1|/usr/bin/v2ray-desktop|' "$SRCDIR"/misc/tpl-linux-autostart.desktop
+
+abinfo "Cutting postfix of Icon..."
```
You are probably talking about "suffix" or "file extension" here.

eatradish committed:
v2ray-desktop: drop

The upstream is no longer maintained and there are problems with availability, so drop
